molgenis-data-migrate/src/main/java/org/molgenis/migrate/version/v1_17/Step27MetaDataAttributeRoles.java

```diff
@@ -146,6 +144,54 @@ public void processRow(ResultSet rs) throws SQLException
 		LOG.info("Updated metadata from version 26 to 27");
 	}
 
+	/**
+	 * Recursive method to find the EntityName for an attribute part which is also a lookup attribute
+	 * @param attrPartName of the attribute part
+	 * @return the entity fullName to which the attribute part belongs
+	 */
+	private String getAttrEntityName(String attrPartName)
+	{
+		String compoundAttrName = getCompoundAttrName(attrPartName);
+		if(compoundAttrIsAttributePart(compoundAttrName))
+		{
+			return getAttrEntityName(compoundAttrName);
+		}
+
+		LOG.info("Finding entity name for attribute: " + compoundAttrName);
+
+		String sql = "SELECT `fullName` FROM `entities_attributes` WHERE `attributes` = '" + compoundAttrName + "'";
+		return jdbcTemplate.queryForObject(sql, new RowMapper<String>()
+		{
+			public String mapRow(ResultSet rs, int rowNum) throws SQLException {
+				return rs.getString("fullName");
+			}
+		});
+	}
+
+	private boolean compoundAttrIsAttributePart(String compoundAttrName)
+	{
+		return getCompoundAttrName(compoundAttrName) != "";
```

Comment:
String is a object, do not use ==/!= to compare strings, use equals (or in this case isEmpty) instead.